org/mozilla/jss/ssl/javax/JSSEngineReferenceImpl.java

```diff
@@ -195,6 +195,12 @@ private void createBufferFD() throws SSLException {
 
         // Initialize ssl_fd from the model Buffer-backed PRFileDesc.
         ssl_fd = SSL.ImportFD(model, fd);
+        if (ssl_fd == null) {
+            throw new SSLException("Error creating SSL socket on top of buffer-backed PRFileDesc.");
+        }
+
+        fd.clear();
```

Comment:
Do we need to call `fd.clear()` if `ssl_fd == null`?

Reply:
We should probably close it, not clear it in that case. I'll update it.